ingestion/src/metadata/ingestion/source/database/redshift/manifest.py

```diff
@@ -0,0 +1,6 @@
+from metadata.ingestion.source.database.redshift.profiler.profiler import (
+    RedshiftProfiler,
+)
+from metadata.utils.manifest import BaseManifest, get_class_path
+
+RedshiftManifest = BaseManifest(profler_class=get_class_path(RedshiftProfiler))
```

Comment:
I am thinking since we control the framework and limit "meta" management of class import and what not when creating additional classes, can we just dynamically import the class based on the service type?

So if I need to add a new class to support some functionality just for Athena, then I just need to implement that class and I don't need to thinking about managing this `manifest.py` file for the specific source.

Reply:
Using a dynamic import approach for all our classes raises to issues:

1. How does one implement a profiler for a [CustomDatabase](https://docs.open-metadata.org/latest/connectors/custom-connectors) in this case?
The idea behind this implementation is to have a default map at the [service type level](https://github.com/open-metadata/OpenMetadata/pull/18322/files#diff-a30dbc8e9bba8105e9857501e0e29627b792608d416e01c65a7aa99af22d2e72R79) and allow the contributors to have more flexibility when developing other modules for their connectors.

2. Our dynamic module resolution is not type safe. One such example is that `BigQuery` is sometimes resolved as `bigquery`, `Bigquery`, or `BigQuery`.  A dynamic import path approach requires the user to have this knowledge and implementing each separate class with these conventions in mind. One can easily forget what is the right casing fora class name and use the wrong conventions when writing the class (for example `BigQuerySource` instead of `BigquerySource`). Without type-safety the only way we can such errors is at runtime. With the current approach there is a single point of failure (the manifest). The connectors themselves are contained within a system that can be validated independently.

Reply:
hi let's stick with strict imports here. It adds a bit of more "boilerplate" when we're developing a new connector but removes magical layers from the code